Jenkinsfile

```diff
@@ -69,6 +72,10 @@ pipeline {
         Lint the code.
         */
         stage('Lint') {
+          when {
+            beforeAgent true
+            expression { return env.ONLY_DOCS == "false" }
```

Comment:
What if we run the docs build first (or close to first) and if we detected it was a docs-only build, we just exited the pipeline right after it run? Would that be less code to maintain or would it make the behavior too rigid? WDYT?